demo/lightning/checkpoint/multinode/fsspecfsdp.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+import gcsfs
+import torch
+
+from pathlib import Path
+from typing import Generator
+from lightning.pytorch.trainer.states import TrainerFn
+from lightning.pytorch.strategies import FSDPStrategy
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+from lightning.pytorch.strategies.fsdp import _METADATA_FILENAME
+from torch.distributed.checkpoint import save, load
+from torch.nn import Module
+from torch.distributed.checkpoint import _fsspec_filesystem as FF
+
+
+class FSSpecFSDPStrategy(FSDPStrategy):
+
+    def __init__(self, path, model, **kwargs):
+        super().__init__(**kwargs)
+        self.model = model
+        self.path = path
+
+    def save_checkpoint(self,
+                        checkpoint,
+                        filepath,
+                        storage_options=None) -> None:
+        if storage_options is not None:
+            raise TypeError(
+                "`FSDPStrategy.save_checkpoint(..., storage_options=...)` is not supported because"
+                " `FSDPStrategy` does not use the `CheckpointIO`.")
+
+        writer = FF.FsspecWriter(self.path, sync_files=False)
+        _ = Path(self.broadcast(filepath))
```

Comment:
What is this doing? Is there a reason we're not using the path but still trying to get the file path?

Reply:
When str "gs://bucket-name" is converted to Path(object) it will get converted to Path(gs:/bucket-name). Since it sort of considers it as NFS path and drops `/`

I faced the same issue long time ago and hence we have this function https://github.com/GoogleCloudPlatform/dataflux-pytorch/blob/main/dataflux_pytorch/lightning/path_utils.py#L10 

So either what Abhi is doing is fine or he will have pass the path via the function linked above. 

But if yeah agreed if he is going to use string path then above code _might be_ unnecessary

Reply:
Replied below. This could just be `self.broadcast(filepath)`.